src/errorpage.cc

```diff
@@ -1352,6 +1352,11 @@ ErrorState::BuildHttpReply()
             request->detailError(type, detail);
         else
             request->detailError(type, SysErrorDetail::NewIfAny(xerrno));
+    } else if (ale) {
+        if (detail)
+            ale->updateError(Error(type, detail));
+        else
+            ale->updateError(Error(type, SysErrorDetail::NewIfAny(xerrno)));
     }
```

Comment:
I am essentially duplicating the xerrno-handling code snippet here. There is already an ErrorState TODO about encapsulating that detail extraction/creation logic:

```C++
    // TODO: Replace type, xerrno and detail with Error while adding a virtual
    // Error::Detail::sysError() method to extract errno in detailError().
    /// type-specific detail about the transaction error;
    /// overwrites xerrno;
    ErrorDetail::Pointer detail;
```

I will address that TODO in a dedicated PR, removing this (and other, similar) code duplication in the process. Unlike this simple bug-fixing PR, that followup refactoring PR may not be suitable for backporting, so I want to get this bug fix (and another ErrorState fix that is waiting for #1368) merged first.